src/data/sparse_page_dmatrix.cc

```diff
@@ -93,7 +83,7 @@ void SparsePageDMatrix::ColPageIter::Init(const std::vector<bst_uint>& index_set
   this->BeforeFirst();
```

Comment:
I still need this because the iterator doesn't know how many columns there are and needs to populate the index set for legacy reasons. I have simplified this function a little more however.

Reply:
Okay, let's keep it for the time being.